src/solvers/dgsem_tree/subcell_limiters.jl

```diff
@@ -52,9 +52,15 @@ function SubcellLimiterIDP(equations::AbstractEquations, basis;
                            positivity_variables_cons = [],
                            positivity_correction_factor = 0.1)
     positivity = (length(positivity_variables_cons) > 0)
-    number_bounds = length(positivity_variables_cons)
 
-    cache = create_cache(SubcellLimiterIDP, equations, basis, number_bounds)
+    bounds_order = Dict()
+    counter = 1
+    for index in positivity_variables_cons
+        bounds_order = Dict(bounds_order..., "$(index)_min" => counter)
```

Comment:
```suggestion
        bounds_order = Dict(bounds_order..., Symbol("$(index)_min") => counter)
```
Always use symbols instead of strings as dict keys. I was looking for a nice explanation online, but I could not find anything satisfactory.

Anyway, one important reason is that symbol comparisons are faster:
```
julia> @btime $"foo" == $"bar";
  4.870 ns (0 allocations: 0 bytes)

julia> @btime $:foo == $:bar;
  2.320 ns (0 allocations: 0 bytes)
```

@sloede Does that still make sense when we never hardcode the symbol and always work with `Symbol(::String)`? Or does it make more sense to make the keys strings in this case?
```
julia> @btime Symbol($"foo") == Symbol($"bar");
  62.172 ns (0 allocations: 0 bytes)

julia> @btime $"foo" == $"bar";
  5.100 ns (0 allocations: 0 bytes)
```

Reply:
@sloede?

Reply:
I created a PR to my complete branch changing the exact same thing as here. See [here](https://github.com/bennibolm/Trixi.jl/pull/114).
There, I show a comparison of runtimes with `Dict` (String and Symbol). Both `Dict` versions seem to add a small amount of allocation in my bounds checking routine. But this is probably a topic for later, since this routine should be revised anyway.

In summary, the version with `Symbol` is almost as fast as the version before. `String` is not that fast. It was hard to benchmark the versions properly since many different routines are affected. So, I just ran an elixir with as many variables as possible.

Reply:
Therefore, I think I will use `Symbol`s as key. Do you agree with this?

Reply:
I am not sure about the performance implications, but what @efaulhaber wrote sounds plausible.

However, as far as I can tell, this is not used in any performance critical part whatsoever, so I think whatever you choose should be ok. The construction of the string using string interpolation is probably more expensive than the string lookup itself (without proof).